ikarus/python/test/testdirichletvalues.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+import debug_info
+
+debug_info.setDebugFlags()
+
+import ikarus as iks
+import dune.grid
+import dune.functions
+
+import math
+
+
+def makeGrid():
+    lowerLeft = []
+    upperRight = []
+    elements = []
+    for _ in range(2):
+        lowerLeft.append(-1)
+        upperRight.append(1)
+        elements.append(3)
+
+    return dune.grid.structuredGrid(lowerLeft, upperRight, elements)
+
+
+def testDirichletValues():
+    grid = makeGrid()
+    basis = iks.basis(grid, dune.functions.Power(dune.functions.Lagrange(order=1), 2))
+    basis = basis.flat()
+
+    dirichletValues = iks.dirichletValues(basis)
+
+    assert basis.size() == dirichletValues.size
+    assert basis.size() == len(dirichletValues)
+
+    def fixOneIndex(vec, globalIndex):
+        if globalIndex == 0:
+            vec[globalIndex] = True
+
+    dirichletValues.fixBoundaryDOFs(fixOneIndex)
+
+    # This is equivalent to values[0] == True, but this syntax is discouraged by PEP
+    assert dirichletValues.container[0]
+
+    # Note that the result of localView.index(localIndex) is a multiIndex even for a flat basis, the localIndex is an int
+    def fixAnotherIndexWithLocalView(vec, localIndex, localView):
+        if localView.index(localIndex) == [4]:
+            vec[localView.index(localIndex)] = True
+
+        assert isinstance(localIndex, int)
+        assert isinstance(localView.index(localIndex), list)
+        assert isinstance(localView.index(localIndex)[0], int)
+
+    dirichletValues.fixBoundaryDOFs(fixAnotherIndexWithLocalView)
+    container = dirichletValues.container
+
+    assert sum(container) == 2
+    assert dirichletValues.fixedDOFsize == 2
+
+    def fixTopSide(vec, localIndex, localView, intersection):
+        if intersection.geometry.center[0] == 1.0:
+            vec[localView.index(localIndex)] = True
+
+        assert isinstance(localIndex, int)
+        assert isinstance(localView.index(localIndex), list)
+        assert isinstance(localView.index(localIndex)[0], int)
+
+    dirichletValues.fixBoundaryDOFs(fixTopSide)
+
+    # This assmues a structured grid
+    indicesPerDirection: int = (math.sqrt(grid.size(0)) + 1) * 2
+    assert dirichletValues.fixedDOFsize == 2 + indicesPerDirection
+
+    # This checks whether container is a reference
+    assert sum(container) == dirichletValues.fixedDOFsize
+
+    # Test Subbasis
+    dirichletValues2 = iks.dirichletValues(basis)
+
+    dirichletValues2.fixBoundaryDOFs(fixOneIndex, 0)
+    dirichletValues2.fixBoundaryDOFs(fixAnotherIndexWithLocalView, 0)
+    dirichletValues2.fixBoundaryDOFs(fixTopSide, 0)
```

Comment:
Is it possible to add a test also corresponding to [fixDOFs()](https://github.com/ikarus-project/ikarus/blob/390f2e106e0a764b86ba6d7803e91dcdd4a9724e/ikarus/python/dirichletvalues/dirichletvalues.hh#L188)?

Reply:
I don't know how to use `fixDOFs` in the Python setting :(

Reply:
We would need something like this, right?
```py
    def fixDOFFunction(basis, vec):
        # forEachDOF(basis)
        pass

    dirichletValues2.fixDOFs(fixDOFFunction)
```

Reply:
Ah is this still open? Overlooked it

Reply:
I would put this in a different PR, I think thats not trivial

Reply:
What do you mean by that? There is `cls.def("fixDOFs",` function defined.

Reply:
https://github.com/ikarus-project/ikarus/pull/305#discussion_r1682399560 this compiles, just don't know what to propoerly do with this

Reply:
> We would need something like this, right?
> 
> ```python
>     def fixDOFFunction(basis, vec):
>         # forEachDOF(basis)
>         pass
> 
>     dirichletValues2.fixDOFs(fixDOFFunction)
> ```

This looks correct? Inside you can just fix one index in vec.

Reply:
Okay sure, but I can only fix the DOFs manually right?

Reply:
I'm sorry I don't get want you mean. It should be the same as for `fixBoundaryDOFs` as in https://github.com/ikarus-project/ikarus/blob/4ba604c9dc4474b1ecda4450b8e8863e17c4bee7/ikarus/python/test/linearelastictest.py#L105-L106

Reply:
But the interface for the method is a different one, you get the basis and the vector. Also we only get one callback and not one for each

Reply:
Yes that's right but this is unrelated to your question, isn't it? But maybe I didn't get your question. You said "I don't know how to use the function in the python setting" and this examples shows how to use it.